spec/integration/autobahn_intg_spec.rb

```diff
@@ -319,6 +319,20 @@ def create_transport_system(name, options={})
         await_delivery
         @queues.map { |q| q.status.first }.reduce(:+).should == 1
       end
+
+      it 'does not deliver any new messages after it has been unsubscribed' do
+        unsubscribed = Autobahn::Concurrency::Semaphore.new(0)
+        @consumer.subscribe do |headers, message|
+          begin
+            unsubscribed.acquire
+            headers.ack
+          rescue java.lang.InterruptedException
+          end
+        end
+        unsubscribed.release
+        @consumer.unsubscribe!
```

Comment:
Given that the block in subscribe is run in another thread `#ack` might be called before `#unsubscribe!`, I can't see how this test ensures that `#ack` is not called either before or after `#unsubscribe!` making it indeterministic?